priv/repo/migrations/20200316103722_create_user_auth_tables.exs

```diff
@@ -1,9 +1,11 @@
-defmodule Demo.Repo.Migrations.CreateAuthTables do
+defmodule Demo.Repo.Migrations.CreateUserAuthTables do
   use Ecto.Migration
 
   def change do
+    execute "CREATE EXTENSION IF NOT EXISTS citext", ""
+
     create table(:users) do
-      add :email, :string, null: false
+      add :email, :citext, null: false
```

Comment:
@aaronrenner @GriffinMB @msaraiva this is an important commit. We need to make the e-mail citext.

@aaronrenner note I have also renamed the migration to something better. The citext command only works for PG, so we would need to detect which adapter the user is running and generate the citext command if they are in PG. I can help with that. :) This is good because if we need to detect for PG, then we can automatically add `async: true` too!

Reply:
@josevalim Any suggestions on the best way to grab the current database adapter? I've been thinking I'd run a regex against the `repo.ex` file and extract the adapter out of there. 
```elixir
defmodule DemoMysql.Repo do
  use Ecto.Repo,
    otp_app: :demo_mysql,
    adapter: Ecto.Adapters.MyXQL
end
```
Is there another approach I should be considering?

Reply:
You should assume there is a repo at "App.Repo", then you can call Code.ensure_loaded?(...) and see if it is loaded, fail otherwise. If it is loaded, the adapter is under `repo.__adapter__`.

Also note all of these checks means you need to call `Mix.Task.run "compile"` at the beginning of the generator. :)

Reply:
Is it worth adding a note explaining this? I've been building my own auth systems for a long time and this escaped me for a lot of that time!

Reply:
what about force the inputted and the persisted `email` to down case on sign in? one way to avoid this specific postgres extension

Reply:
The extension is by far the best and more efficient approach and makes it consistent with other DBs too, which are case insensitive by default.